lib/calc_route.dart

```diff
@@ -0,0 +1,137 @@
+import 'dart:math';
+import 'package:args/args.dart';
+import 'cart_point.dart';
+import 'route_model.dart';
+
+import 'package:sail_routing_dart/route_writer.dart';
+
+bool v_flag = false;
+
+void main(List<String> args) {
+  var parser = new ArgParser();
+  parser.addFlag("verbose", abbr: 'v', defaultsTo: false);
+  var results = parser.parse(args);
+  v_flag = results['verbose'];
+
+  Cart_Point start = new Cart_Point(0.0, 0.0);
+  Cart_Point end = new Cart_Point(1.0, -10.2);
+  double wind = 3 * pi / 2;
+  RouteModel route = new RouteModel(start: start, end: end, wind_radians: wind);
+
+  calculateOptimal(route);
+  print(route);
+
+  RouteWriter rw = new RouteWriter();
+  rw.writeToFileFromRouteModel(route: route);
+  rw.run_python_plotter();
+}
+
+double euc_dist(Cart_Point a, Cart_Point b) {
+  return a.toPoint().distanceTo(b.toPoint());
+}
+
+void verbose_print(dynamic s) {
+  if (v_flag) {
+    print(s);
+  }
+}
+
+Cart_Point find_intersection(Cart_Point a_point, Cart_Point b_point, Cart_Point c_point, Cart_Point d_point) {
+  double a1 = b_point.y - a_point.y;
+  double b1 = a_point.x - b_point.x;
+  double c1 = a1 * (a_point.x) + b1 * (a_point.y);
+
+  // Line CD represented as a2x + b2y = c2
+  double a2 = d_point.y - c_point.y;
+  double b2 = c_point.x - d_point.x;
+  double c2 = a2 * (c_point.x) + b2 * (c_point.y);
+  double determinant = a1 * b2 - a2 * b1;
+
+  if (determinant == 0) {
+    print("find_intersection() found parallel");
+    throw Error();
+  } else {
+    double x = (b2 * c1 - b1 * c2) / determinant;
+    double y = (a1 * c2 - a2 * c1) / determinant;
+
+    verbose_print("intersection: " + Cart_Point(x, y).toString());
+
+    return new Cart_Point(x, y);
+  }
+}
+
+bool check_viable_tack(Cart_Point p1, Cart_Point p2, double wind_radians) {
```

Comment:
https://www.dartlang.org/guides/language/effective-dart/style I think lowerCamelCase for check_viable_tack